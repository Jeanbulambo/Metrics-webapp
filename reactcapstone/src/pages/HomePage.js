import { useSelector } from 'react-redux';
import React, { useEffect, useState } from 'react';
import { IoMdMic } from 'react-icons/io';
import { FiSettings } from 'react-icons/fi';
import './styles/Home.scss';
import { getMapUrl } from '../assets/Urlmapping';
import CountriesList from '../components/CountriesList';

const HomePage = () => {
  const { status, data } = useSelector((state) => state.Countries);
  const visibleData = data.filter((country) => country.id !== undefined);
  const [search, setSearch] = useState(visibleData);

  useEffect(() => {
    setSearch(visibleData);
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [data]);

  const filterData = (e) => {
    const string = e.target.value.toLowerCase();
    const info = visibleData.filter((country) => country.id.toLowerCase().includes(string));
    if (info !== 0) {
      setSearch(info);
    } else {
      setSearch(visibleData);
    }
  };

  const worldMap = getMapUrl('world');

  return (
    <>
      <div className="header-container">
        <div className="ouline-left">
          <h3>2022</h3>
        </div>
        <h4 className="center-title" data-testid="title">most views</h4>
        <div className="header-buttons">
          <button type="button" className="microphone" label="microphone"><IoMdMic /></button>
          <button type="button" className="settings" label="Settings" title="settings"><FiSettings /></button>
        </div>
      </div>
      { status === 'success' && (
      <>
        <div className="firts-card">
          <div className="map-container">
            <img src={worldMap} alt="World Map" className="map" />
          </div>
          <div>
            <h2>COVID-19 </h2>
            <p>Confirm Cases</p>
            <input className="input" type="text" name="search" onChange={filterData} placeholder="Search by country" />
          </div>
        </div>
        <div className="stast">
          STATS BY COUNTRY
        </div>
        <div>
          { search === ''
            ? (
              <div className="countries-container">
                <CountriesList countries={visibleData} />
              </div>
            ) : (<CountriesList countries={search} />)}
        </div>
      </>
      )}
    </>
  );
};

export default HomePage;
