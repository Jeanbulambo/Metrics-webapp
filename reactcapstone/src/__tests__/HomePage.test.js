import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../Redux/configureStore';
import HomePage from '../pages/HomePage';

it('Check if HomePage page renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <HomePage />
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});

afterEach(() => {
  jest.clearAllMocks();
});
