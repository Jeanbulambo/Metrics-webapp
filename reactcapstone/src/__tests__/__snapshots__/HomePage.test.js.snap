// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if HomePage page renders correctly 1`] = `
<div
  className="header-container"
>
  <div
    className="ouline-left"
  >
    <h3>
      2022
    </h3>
  </div>
  <h4
    className="center-title"
    data-testid="title"
  >
    most views
  </h4>
  <div
    className="header-buttons"
  >
    <button
      className="microphone"
      label="microphone"
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 320c37.712 0 68.571-30.924 68.571-68.714V100.714C324.571 62.924 293.712 32 256 32s-68.571 30.924-68.571 68.714v150.572c0 37.79 30.859 68.714 68.571 68.714zm121.139-75.452c0 68.714-58.282 116.815-121.139 116.815s-121.139-48.102-121.139-116.815H96c0 77.873 61.719 143.153 137.144 153.465V480h45.713v-81.987C354.281 386.561 416 322.421 416 244.548h-38.861z"
        />
      </svg>
    </button>
    <button
      className="settings"
      label="Settings"
      title="settings"
      type="button"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="3"
        />
        <path
          d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
        />
      </svg>
    </button>
  </div>
</div>
`;
